### PR TITLE
Add support for link flag allmulticast

### PR DESCRIPTION
**Pull request by pmorjan**
I need to set the allmulticast flag on Macvtap interfaces.
This is equivalent to: ip link set <link> allmulticast <on|off>

### CODE REVIEW

File: link_test.go [hunk 1]
```diff
@@ -1869,3 +1869,46 @@ func TestLinkSetBondSlave(t *testing.T) {
...
+		t.Fatal(err)
+	}
+
+	link, err = LinkByName("foo")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if link.Attrs().RawFlags&unix.IFF_ALLMULTI != uint32(unix.IFF_ALLMULTI) {
+		t.Fatal("ALLMUILTI was not set!")
```
Comment (fcrisciani) [inline link_test.go]: ALLMULTI?

File: link_test.go [hunk 2]
```diff
@@ -1869,3 +1869,46 @@ func TestLinkSetBondSlave(t *testing.T) {
...
+		t.Fatal(err)
+	}
+
+	link, err = LinkByName("foo")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if link.Attrs().RawFlags&unix.IFF_ALLMULTI != 0 {
+		t.Fatal("ALLMUILTI is still set!")
```
Comment (fcrisciani) [inline link_test.go]: same typo here I guess

Comment (pmorjan) [reply link_test.go]: yes, fixed.

File: link_linux.go [hunk 3]
```diff
@@ -113,6 +113,52 @@ func (h *Handle) SetPromiscOn(link Link) error {
...
+// LinkSetAllmulticastOff disables the reception of all hardware multicast packets for the link device.
+// Equivalent to: `ip link set $link allmulticast off`
+func (h *Handle) LinkSetAllmulticastOff(link Link) error {
+	base := link.Attrs()
+	h.ensureIndex(base)
+	req := h.newNetlinkRequest(unix.RTM_NEWLINK, unix.NLM_F_ACK)
+
+	msg := nl.NewIfInfomsg(unix.AF_UNSPEC)
+	msg.Change = unix.IFF_ALLMULTI
+	msg.Flags = 0 & ^unix.IFF_ALLMULTI
```
Comment (fcrisciani) [inline link_linux.go]: is't this the same as `msg.Flags = unix.IFF_ALLMULTI`?

Comment (pmorjan) [reply link_linux.go]: yes that's stupid, need to find the correct value. Thanks.

Comment (pmorjan) [reply link_linux.go]: Isn't  SetPromiscOff and LinkSetDown the same?
https://github.com/pmorjan/netlink/blob/23bf27b4a71ebda39cca31071aac54a7217503f6/link_linux.go#L251-L252
https://github.com/pmorjan/netlink/blob/23bf27b4a71ebda39cca31071aac54a7217503f6/link_linux.go#L301-L302

Comment (fcrisciani) [reply link_linux.go]: they seems so, but not sure why all that extra care is needed honestly. @aboch @vishvananda wdyt?

Comment (aboch) [reply link_linux.go]: @jellonek @pmorjan Feel free to push a patch to fix it.

**Pull request merged**